[visionOS] REGRESSION: Seeking WebKit streamed MSE videos result in playback error

jyavenard opened the pull request:
#### c555dcb9a1c3f3c5738bd7f83733520ec9049531
<pre>
[visionOS] REGRESSION: Seeking WebKit streamed MSE videos result in playback error
<a href="https://bugs.webkit.org/show_bug.cgi?id=291517">https://bugs.webkit.org/show_bug.cgi?id=291517</a>
<a href="https://rdar.apple.com/148909355">rdar://148909355</a>

Reviewed by Youenn Fablet.

Ignore kVTVideoDecoderReferenceMissingErr error returned by a VT Session.

When seeking following a (Coded Random Access) CRA frames, a VT Session may
enter a condition that will cause to return an error. The error is recoverable.

Manually tested on vision pro.
Adding an automated test has proven to be difficult because:
1- Those MV-HEVC videos can only be decoded on a real device.
2- The original content reproducing the issue is of significant size (85MiB)
3- The simple act of seeking isn&apos;t sufficient to reproduce the error, you
   need to seek after playing a certain amount of frames, amount that hasn&apos;t been perfectly identified.

* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm:
(WebCore::WebCoreDecompressionSession::isNonRecoverableError const):
(WebCore::WebCoreDecompressionSession::handleDecompressionOutput):

Canonical link: <a href="https://commits.webkit.org/293692@main">https://commits.webkit.org/293692@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/284c15133790b1bbd921afb0065abb5740007316

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99638 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19288 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9549 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104769 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50234 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101679 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19577 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27721 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/75844 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32943 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102645 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14906 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89968 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/56203 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14705 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7954 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49597 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/84670 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8039 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107128 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26753 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/19530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/84803 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27117 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86172 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84322 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21404 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28996 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6701 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20543 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26693 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31895 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26508 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29824 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28078 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm:
```diff
@@ -290,16 +290,21 @@
 
-    if (status != noErr) {
+    if (isNonRecoverableError(status)) {
         m_lastDecodingError = status;
```
Should we add logging?

jyavenard committed:
[visionOS] REGRESSION: Seeking WebKit streamed MSE videos result in playback error
https://bugs.webkit.org/show_bug.cgi?id=291517
rdar://148909355

Reviewed by Youenn Fablet.

Ignore kVTVideoDecoderReferenceMissingErr error returned by a VT Session.

When seeking following a (Coded Random Access) CRA frames, a VT Session may
enter a condition that will cause to return an error. The error is recoverable.

Manually tested on vision pro.
Adding an automated test has proven to be difficult because:
1- Those MV-HEVC videos can only be decoded on a real device.
2- The original content reproducing the issue is of significant size (85MiB)
3- The simple act of seeking isn't sufficient to reproduce the error, you
   need to seek after playing a certain amount of frames, amount that hasn't been perfectly identified.

* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm:
(WebCore::WebCoreDecompressionSession::isNonRecoverableError const):
(WebCore::WebCoreDecompressionSession::handleDecompressionOutput):

Canonical link: https://commits.webkit.org/293692@main
